app/api/v1/users.rb

```diff
@@ -0,0 +1,19 @@
+class V1::Users < Grape::API
+  
+  before do
+    authenticate!
+    
+    @allowed_users = current_team&.users || User.all
```

Comment:
If the token isn't linked to any team, it means that the token is global and have full access to see this data. Otherwise, the token is team related and only the members of the team's token is listed. It will list nothing if no token or a invalid token is given.